dts/bindings/misc/renesas,ra-agt.yaml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2024 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Renesas RA AGT
+
+compatible: "renesas,ra-agt"
+
+include: base.yaml
+
+properties:
+  reg:
+    required: true
+
+  channel:
+    type: int
+    required: true
+
+  count-source:
```

Comment:
Why not use the clock's PHANDLE?
DeviceTree's descriptability should usually be prioritized over Ease of implementation.

Reply:
@soburi , This configuration is for AGT count source, in this case the configuration is not only to support for clock device but also other external signal.
![image](https://github.com/user-attachments/assets/ef1f2270-7664-43e7-95bf-f4502aa3c7ee)

Reply:
@soburi , Currently this PR is aim to only support 2 count source which is PCLKB and LOCO, other will be support later, the explanation is to aware of this property is not for clock device configration.

Reply:
It may no need to be blocked. But it should be better to describe why the features are not implemented.

Reply:
@soburi , would it be appropriate to add them as comment like this?

```suggestion
  count-source:
  description: AGT count source.
  type: string
  required: true
  enum:
    - "AGT_CLOCK_PCLKB"
    - "AGT_CLOCK_LOCO"
    # Below count source configuration will be supported later
    # - "AGT_CLOCK_AGT_UNDERFLOW"
    # - "AGT_CLOCK_SUBCLOCK"
    # - "AGT_CLOCK_P402"
    # - "AGT_CLOCK_P403"        
    # - "AGT_CLOCK_P404"        
    # - "AGT_CLOCK_AGTIO"        
```

Reply:
I think it's ok to include it in the commit message.
It intends only to clear the reason.

Reply:
I updated the commit message for this support